tools/gmon/gmon/cli.py

```diff
@@ -0,0 +1,425 @@
+    fields = getattr(args, 'fields', None)
+    fields = parse_fields(fields)
+    filters = getattr(args, 'filters', [])
+    filters = parse_filters(filters)
+
+    if parser == 'metrics':
+        client = MetricsClient(args.project)
+        method = getattr(client, command)
+
+        if command in ['get', 'delete']:
+            response = method(metric_type)
+
+        elif command in ['create']:
+            response = method(metric_type, args.metric_kind, args.value_type,
+                              args.description)
+
+        elif command in ['list']:
+            response = method(pattern=args.regex, fields=fields)
+
+        elif command in ['inspect']:
+            response = method(metric_type, window=args.window)
+
+        elif command in ['delete_unused']:
+            response = method(pattern=args.regex, window=args.window)
+
+        # Format, filter and print response
+        print_response(response, limit, fields, filters)
+
+    elif parser == 'accounts':
+        if 'project' in args:
+            client = AccountClient(project_id=args.project,
+                                   no_poll=args.no_poll)
+        else:
+            client = AccountClient()
+        method = getattr(client, command)
+        if command in ['get', 'create', 'delete', 'list']:
+            response = method()
+        elif command in ['link', 'unlink']:
+            response = method(project_id=args.project_id)
+
+    elif parser == 'services':
+        client = ServiceMonitoringClient(project_id=args.project)
+        if command in ['get', 'create', 'delete']:
+            method = getattr(client, command + '_service')
+            response = method(args.service_id)
+        else:
+            method = getattr(client, command + '_services')
+            response = method()
+        print_response(response, limit, fields, filters)
+
+    elif parser == 'slos':
+        client = ServiceMonitoringClient(project_id=args.project)
+        if command in ['get', 'create', 'delete']:
+            method = getattr(client, command + '_slo')
+            response = method(args.service_id, args.slo_id)
+        else:
+            method = getattr(client, command + '_slos')
+            response = method(args.service_id)
+        print_response(response, limit, fields, filters)
+
+
+def parse_filters(filters=[]):
+    """Function to parse `filters` CLI argument.
+
+    Args:
+        filters: A list of "key=value" strings.
+
+    Returns:
+        list: Parsed filters
+    """
+    ret = {}
+    for f in filters:
+        k, v = f.split("=")
+        ret[k] = v
+    return ret
+
+
+def filter_response(response, filters={}):
+    """Filter response from Stackdriver Monitoring APIs.
+
+    Args:
+        response (dict): JSON response.
+        filters (dict): Filters as key-value dict.
+
+    Returns:
+        dict: Filtered response.
+    """
+    if not filters:
+        return response
+    dotted = DotMap(response)
+    for k, v in filters.items():
```

Comment:
these filters can't be pushed server side into SD API request?

Reply:
unfortunately no, the client doesn't allow filtering the response fields (unlike `gcloud`)